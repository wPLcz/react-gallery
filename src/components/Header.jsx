import React, { useEffect } from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';
import Button from "./Button";

const Header = (props) => {
  return (
    <Section>
      <Title>{props.title}</Title>
      <ButtonsContainer>
        {Object.keys(props.options).map(sortingOption => (
            <Button
              active={props.options[sortingOption]}
              key={uuidv4()}
              id={sortingOption}
              takeNewestSorting={(id) => props.takeNewestSorting(id)}>
              {sortingOption}
            </Button>
          )
        )}
      </ButtonsContainer>
    </Section>
  )
};

const Title = styled.h2`
    display: block;
    text-align: center;
    font-size: 32px;
    max-height: 25vh;
    border-radius: 3px;
    font-weight: 200;
    font-family: Helvetica;
    text-transform: uppercase;
    color: #999;
    
    @media (min-width: 960px) {
      margin: 0 auto;
      display: inline-block;
      font-size: 88px;
      margin: 0 0 0 2vw;
      text-align: left;
    }
`;

const Section = styled.section`
    margin: 5vh 0;
    width: 100%;
    max-height: 25vh;  
`;

const ButtonsContainer = styled.div`
    display: flex;
    justify-content: center;
    
    @media (min-width: 960px) {
      justify-content: flex-start;
      margin: 0 0 0 2%;
    }
`;

Header.propTypes = {
  options: PropTypes.object,
  title: PropTypes.string,
  click: PropTypes.func,
};

Header.defaultProps = {
  options: {},
  title: '',
  click: () => {
  },
};

export default Header;
