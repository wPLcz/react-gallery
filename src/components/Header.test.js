import React from "react";
import ReactDOM from "react-dom";
import renderer from 'react-test-renderer';
import Header from "./Header";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Header/>, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders correctly when there are no items', () => {
  const tree = renderer.create(<Header/>).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly passing title', () => {
  const tree = renderer.create(<Header title={"Mocked Title"}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
