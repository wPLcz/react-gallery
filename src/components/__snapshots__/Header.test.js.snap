// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly passing title 1`] = `
<section
  className="sc-dlnjPT eIaXWU"
>
  <h2
    className="sc-gtssRu lmdLgS"
  >
    Mocked Title
  </h2>
  <div
    className="sc-hKFyIo exTVht"
  />
</section>
`;

exports[`renders correctly when there are no items 1`] = `
<section
  className="sc-dlnjPT eIaXWU"
>
  <h2
    className="sc-gtssRu lmdLgS"
  >
    
  </h2>
  <div
    className="sc-hKFyIo exTVht"
  />
</section>
`;
